### ISSUE TITLE
PublicApiAnalyzers report spurious diagnostics around value types, blocking upgrade

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn-analyzers#7375]
### Analyzer

**Diagnostic ID**: RS0036 missing nullability annotations

**Diagnostic ID**: RS0041 oblivious types

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.PublicApiAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.PublicApiAnalyzers)

**Version**: 3.11.0-beta1.24324.1

### Describe the bug

[Updating the PublicAPI analyzers version](https://github.com/microsoft/vs-threading/compare/main...AArnott:syncAnalyzerVersions) in the vs-threading repo has led to many build warnings that cannot be resolved.

### Steps To Reproduce

```ps1
git clone https://github.com/aarnott/vs-threading
cd vs-threading
git checkout syncAnalyzerVersions
.\init.ps1 # or init.cmd
dotnet build src\Microsoft.VisualStudio.Threading
```

### Expected behavior

No warnings, or at least only warnings that make sense and can be resolved.

### Actual behavior

About 60 warnings _per target framework_, most or all of which are either:

* RS0036 diagnostics complaining about missing nullability annotations
  > C:\Users\andarno\source\repos\vs-threading\src\Microsoft.VisualStudio.Threading\AsyncReaderWriterResourceLock`2.cs(386,33): warning RS0036: Symbol 'Microsoft.VisualStudio.Threading.AsyncReaderWriterResourceLock<TMoniker, TResource>.ResourceAwaiter.GetResult() -> Microsoft.VisualStudio.Threading.AsyncReaderWriterResourceLock<TMoniker, TResource>.ResourceReleaser' is missing nullability annotations in the declared API (https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md) [C:\Users\andarno\source\repos\vs-threading\src\Microsoft.VisualStudio.Threading\Microsoft.VisualStudio.Threading.csproj::TargetFramework=net472]

  But the odd thing here is they are all regarding _structs_ for which nullable reference annotations don't make sense.
* RS0041 diagnostics complaining about oblivious types:
  > C:\Users\andarno\source\repos\vs-threading\src\Microsoft.VisualStudio.Threading\AsyncReaderWriterResourceLock`2.cs(137,34): warning RS0041: Symbol 'UpgradeableReadLockAsync' uses some oblivious reference types (https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md) [C:\Users\andarno\source\repos\vs-threading\src\Microsoft.VisualStudio.Threading\Microsoft.VisualStudio.Threading.csproj::TargetFramework=net472]

  This too doesn't seem to be addressible.

**Comment by AArnott** [comment]
This is still a blocking issue.

**Comment by AArnott** [comment]
All of the failures happen around types that are nested under a generic type. So a minimal repro may be something like:

```cs
public class A<T>
{
   public enum { A }
}
```
